snapshot/snapshotfs/upload.go

```diff
@@ -201,6 +201,43 @@ func (u *Uploader) uploadSymlinkInternal(ctx context.Context, relativePath strin
 	return de, nil
 }
 
+func (u *Uploader) uploadStreamingFileInternal(ctx context.Context, relativePath string, f fs.StreamingFile) (*snapshot.DirEntry, error) {
+	u.Progress.HashingFile(relativePath)
+	defer u.Progress.FinishedHashingFile(relativePath, f.Size())
+
+	reader, err := f.GetReader(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to get streaming file reader")
+	}
+
+	writer := u.repo.NewObjectWriter(ctx, object.WriterOptions{
+		Description: "STREAMFILE:" + f.Name(),
+	})
+	defer writer.Close() //nolint:errcheck
+
+	written, err := u.copyWithProgress(writer, reader, 0, f.Size())
```

Comment:
f.Size() is not known (likely zero). What does this do in terms of progress output?

Reply:
I see something like this in logs:
```
 \ 1 hashing, 0 hashed (4.9 MB), 0 cached (0 B), uploaded 210 B, estimating...[STORAGE] PutBlob("s64734ac1f42cc0e46939dc1aca1945c6-s092fe2d9a70f48d6101",len=210)=<nil> took 536.342µs
 - 1 hashing, 0 hashed (27.7 MB), 0 cached (0 B), uploaded 27.7 MB, estimating...[STORAGE] PutBlob("pd0f636a786684f59e675c87f712d91a7-s092fe2d9a70f48d6101",len=27738411)=<nil> took 11.28855ms
 - 1 hashing, 0 hashed (52.8 MB), 0 cached (0 B), uploaded 52.9 MB, estimating...[STORAGE] PutBlob("pb16aff7cb127b03d47135d0a0689b163-s092fe2d9a70f48d6101",len=25170182)=<nil> took 12.801101ms
 | 1 hashing, 0 hashed (80.6 MB), 0 cached (0 B), uploaded 80.7 MB, estimating...[STORAGE] PutBlob("p612a62fa2d2fc809f1e6d019a30e6c01-s092fe2d9a70f48d6101",len=27750699)=<nil> took 15.916912ms
 / 1 hashing, 0 hashed (104.9 MB), 0 cached (0 B), uploaded 104.9 MB, estimating...[STORAGE] PutBlob("p13f4a34665dfbcc1eebfb0fa39924a8a-s092fe2d9a70f48d6101",len=24232198)=<nil> took 17.461747ms
 / 1 hashing, 0 hashed (130 MB), 0 cached (0 B), uploaded 130.1 MB, estimating...[STORAGE] PutBlob("p7f7afdc67fe4fcf6f7dbeeaa9296afd4-s092fe2d9a70f48d6101",len=25170182)=<nil> took 75.648642ms
 - 1 hashing, 0 hashed (154.3 MB), 0 cached (0 B), uploaded 154.4 MB, estimating...[STORAGE] PutBlob("p69ac3031aeff454355b323bb3ccd62a6-s092fe2d9a70f48d6101",len=24297771)=<nil> took 15.973844ms
 / 1 hashing, 0 hashed (179.5 MB), 0 cached (0 B), uploaded 179.5 MB, estimating...[STORAGE] PutBlob("pf46dc28ab08fa923ca28880bc1b5d409-s092fe2d9a70f48d6101",len=25170182)=<nil> took 20.798219ms
 / 1 hashing, 0 hashed (200.9 MB), 0 cached (0 B), uploaded 200.9 MB, estimating...[STORAGE] PutBlob("p1b9c119c0c69548d0b81f58fd8c4f18e-s092fe2d9a70f48d6101",len=21414150)=<nil> took 9.483296ms
 - 1 hashing, 0 hashed (200.9 MB), 0 cached (0 B), uploaded 200.9 MB, estimating...listing manifest contents
  keeping 2021-03-04 00:11:11.828543 -0800 PST reasons: [latest-1,annual-1,monthly-1,weekly-1,daily-1,hourly-1]
 \ 0 hashing, 1 hashed (201 MB), 0 cached (0 B), uploaded 201 MB, estimating...    [STORAGE] PutBlob("pb0ca7b2793316116d82236c202ebd69e-s092fe2d9a70f48d6101",len=86203)=<nil> took 443.026µs
 ```